content/doc/administration/requirements/upgrade-java-guidelines.adoc

```diff
@@ -0,0 +1,39 @@
+---
+layout: documentation
+title: Upgrading Jenkins Java version from 8 to 11
+---
+
+In order to upgrade the JVM used to run Jenkins, more specifically from Java 8 to Java 11, there are a few details and steps to take care.
+
+== Backup
+
+As for any upgrade, we recommend you to backup the `JENKINS_HOME`.
+It is also recommended to test the upgrade with the backup before performing it on the production instance.
+
+== Upgrading Plugins
+
+Some plugins were not previously compatible with Java 11.
+Therefore it is important that you upgrade all of your plugins.
+
+One of the most important upgrades is the plugin:workflow-support[Workflow Support plugin].
+We need to make sure that the version of the plugin is at least `3.0`.
+Because this upgrade changes the serialization of the Pipeline builds, it requires that no Pipeline jobs are running when upgrading this plugin. 
+
+Even if Pipeline jobs are supposed to survive a Jenkins restart, when the restart of the instance is due to a Pipeline related plugin, it is always a better option to make sure that no Pipeline builds are in progress before restarting the instance.
+
+== JAXB libraries
+
+Some plugins were using JAXB libraries, provided by the JDK running Jenkins.
+Those libraries are no longer provided, since Java 9.
+
+In order to fix this, we bundled those libraries in a new detached plugin: plugin:jaxb[JAXB plugin].
+This new plugin will be automatically installed with Jenkins core more recent than `2.163`.
+
+== JVM version on agents
+
+It is important to know that because of the communication system between the master and its agents, all agents must be running on the same JVM version as the master.
+This means, as we are about to run Jenkins on Java 11, we need to make sure the agents JVM are upgraded as well.
```

Comment:
Shouldn't we mention something about agents launched via JWS?

Reply:
good point.